### PR TITLE
[Docs] Add dataframe, Snowflake, Azure examples, and minor text edits

### PR DESCRIPTION
**Pull request by jillnogold**

### CODE REVIEW

File: docs/feature-store/using-spark-engine.md [hunk 1]
```diff
@@ -238,4 +267,72 @@ target = ParquetTarget(
...
+
+fn = code_to_function(kind='remote-spark',  name='func')
+
+run_config = fstore.RunConfig(local=False, function=fn, handler="ingest_handler")
+
+target = ParquetTarget(partitioned = True, time_partitioning_granularity="month")
+
+feature_set.set_targets(targets=[target],with_defaults=False)
+
+fstore.ingest(feature_set, source, run_config=run_config, spark_context=spark_service_name)
```
Comment (theSaarco) [inline docs/feature-store/using-spark-engine.md:338]: This example is missing Azure credentials setting. The code should either show how to set credentials (like in the S3 example) or at least note that it's assumed that the correct env variables were set using project-secrets beforehand. This is the same as in the S3 example above.

**Review by theSaarco — CHANGES_REQUESTED** [review]
Reviewed - some items still need work, especially note the Azure example which has some issues in it.

File: docs/feature-store/using-spark-engine.md [hunk 2]
```diff
@@ -160,9 +165,33 @@ config = fstore.RunConfig(local=False, function=my_func, handler="ingest_handler
...
+fset = fstore.FeatureSet("myset", entities=[fstore.Entity("id")], engine="spark")
+
+fstore.ingest(fset, df, spark_context=spark)
+
+spark.stop()
+```
+
+## Spark over S3 - full flow example
+
+For Spark to work with S3, it requires several properties to be set. Spark over S3 can be executed both remotely and locally. The following example writes a
```
[the remark this answers is not in the document]
Comment (jillnogold) [reply docs/feature-store/using-spark-engine.md]: added

File: docs/feature-store/using-spark-engine.md [hunk 3]
```diff
@@ -238,4 +268,73 @@ target = ParquetTarget(
...
+run_config = fstore.RunConfig(local=False, function=fn, handler="ingest_handler")
+
+target = ParquetTarget(partitioned = True, time_partitioning_granularity="month")
+
+feature_set.set_targets(targets=[target],with_defaults=False)
+
+fstore.ingest(feature_set, source, run_config=run_config, spark_context=spark_service_name)
+
+import mlrun
+import mlrun.feature_store as fs
```
Comment (theSaarco) [inline docs/feature-store/using-spark-engine.md:339]: I think you misread my previous comment - I just meant for the code markers to be at the end of code. These lines belong at the beginning of the code. However, as noted in other comments - the 2nd line is redundant, just move the `import mlrun` line to the beginning.

File: docs/feature-store/using-spark-engine.md [hunk 4]
```diff
@@ -238,4 +268,73 @@ target = ParquetTarget(
...
+feature_set = fstore.FeatureSet(
+    name="customer", entities=[fstore.Entity("C_CUSTKEY")], engine="spark"
+)
+
+source = SnowflakeSource(
+    "customer_sf",
+    query="select * from customer limit 100000",
+    url="<url>",
+    user="nnnnnn",
+    password="nnnnnn",
```
Comment (theSaarco) [inline docs/feature-store/using-spark-engine.md:296]:
```suggestion
    user="<username>",
    password="<password>",
```

If we're using it for the `<url>` parameter, might just as well stick to the convention.

File: docs/feature-store/using-spark-engine.md [hunk 5]
```diff
@@ -238,4 +268,73 @@ target = ParquetTarget(
...
+project_name = "spark-azure-test"
+project = mlrun.get_or_create_project(project_name, context="./")
+
+from mlrun.runtimes import RemoteSparkRuntime
+RemoteSparkRuntime.deploy_default_image()
+
+from mlrun.datastore.sources import CSVSource
+from mlrun.datastore.targets import ParquetTarget
+from mlrun import code_to_function
+import mlrun.feature_store as fstore
```
Comment (theSaarco) [inline docs/feature-store/using-spark-engine.md:320]: This line is redundant assuming we return the `import mlrun.feature_store as fs` line to where it belongs. 
However, My suggestion is to do the following:
* Only keep this line (remove the `import mlrun.feature_store as fs` line altogether)
* Make sure only `fstore.` is used - there's a single place to change - see next comment.

File: docs/feature-store/using-spark-engine.md [hunk 6]
```diff
@@ -238,4 +268,73 @@ target = ParquetTarget(
...
+
+feature_set = fs.FeatureSet("rides7", entities=[fs.Entity("ride_id")], engine="spark", timestamp_key="key")
```
Comment (theSaarco) [inline docs/feature-store/using-spark-engine.md:322]:
```suggestion
feature_set = fstore.FeatureSet("rides7", entities=[fstore.Entity("ride_id")], engine="spark", timestamp_key="key")
```

**Review by theSaarco — CHANGES_REQUESTED** [review]
Almost there - a few more changes needed.

**Comment by jillnogold** [comment]
@theSaarco - good to go now?

**Review by theSaarco — APPROVED** [review]
LGTM now.

**Review by Hedingber — APPROVED** [review]

**Pull request merged**